src/libostree/ostree-sysroot-deploy.c

```diff
@@ -2994,6 +3040,9 @@ ostree_sysroot_write_deployments_with_options (OstreeSysroot *self, GPtrArray *n
         return FALSE;
     }
 
+  if (new_deployments->len && !kexec_load_kernel(self, new_deployments->pdata[0], cancellable, error))
```

Comment:
This needs to be conditional on the flag, right?

I think that's why the unit tests are all dying (because of lack of privileges for the syscall).

But anyways if we follow my suggestion of...mmm, a new C API like `ostree_sysroot_deployment_kexec_load()` with a thin new CLI verb, then that issue would go away.

Reply:
Oops. Thanks, good catch.